lib/src/modules/conditional.dart

```diff
@@ -831,7 +831,12 @@ class ConditionalAssign extends Conditional {
       guard(driver);
     }
 
-    _receiverOutput.put(driverValue(driver));
+    final currentValue = driverValue(driver);
+    if (!currentValue.isValid) {
+      _receiverOutput.put(LogicValue.x);
+    } else {
+      _receiverOutput.put(currentValue);
+    }
 
     if (!drivenSignals.contains(receiver) || receiver.value.isValid) {
       drivenSignals.add(receiver);
```

Comment:
I think there's another case in the `If` where we need to make sure to prop a X instead of Z:
```dart
      } else if (driverValue(iff.condition) != LogicValue.zero) {
        // x and z propagation
        for (final receiver in receivers) {
          receiverOutput(receiver).put(driverValue(iff.condition)[0]);
          if (!drivenSignals.contains(receiver) || receiver.value.isValid) {
            drivenSignals.add(receiver);
          }
        }
        break;
      }
```